File: prime/src/test/java/org/ostelco/prime/storage/entities/ProductTest.java
package org.ostelco.prime.storage.entities;

import org.junit.Test;
import org.ostelco.prime.storage.Products;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public final class ProductTest {

    private static final String SKU = "SKU-1";

    private static final String DESCRIPTION = "a random description";

    private final Product product = new Product(SKU, DESCRIPTION);


    @Test(expected = NotATopupProductException.class)
    public void asTopupProductNot() throws NotATopupProductException {
        product.asTopupProduct();
    }

    public void asTopupProductTrue() throws NotATopupProductException {
        // Ghetto, not proper testing.
        assertTrue(Products.getProductForSku("DataTopup3GB").asTopupProduct()
                instanceof  TopUpProduct);
    }

    @Test
    public void isTopUpProject() {
        assertFalse(product.isTopUpProject());
    }
}

File: prime/src/test/kotlin/org/ostelco/prime/storage/entities/ProductTest.kt
package org.ostelco.prime.storage.entities

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import org.ostelco.prime.model.Product
import org.ostelco.prime.model.TopUpProduct
import org.ostelco.prime.storage.Products

class ProductTest {

    private val product = Product(SKU, DESCRIPTION)


    @Test(expected = NotATopupProductException::class)
    @Throws(NotATopupProductException::class)
    fun asTopupProductNot() {
        product.asTopupProduct()
    }

    @Throws(NotATopupProductException::class)
    fun asTopupProductTrue() {
        // Ghetto, not proper testing.
        assertTrue(Products.getProductForSku("DataTopup3GB")!!.asTopupProduct() is TopUpProduct)
    }

    @Test
    fun isTopUpProject() {
        assertFalse(product.isTopUpProject())
    }

    companion object {

        private const val SKU = "SKU-1"

        private const val DESCRIPTION = "a random description"
    }
}
